projects/hslayers/src/components/styles/styler.service.ts

```diff
@@ -751,4 +783,111 @@ export class HsStylerService {
       this.hsLogService.warn('SLD could not be parsed', err);
     }
   }
+
+  /**
+   * Calculate a stop-color value for gradient. Offset depend on number of colors (lenght)
+   * and current index.
+   * @param color RGB(A) color definition
+   * @param linearGradient SVG Linear Gradient element
+   * @param length Offset calculation - length of an array (number of colors)
+   * @param index Index of color
+   */
+  private appendColorToGradient(
+    color: number[],
+    linearGradient: SVGLinearGradientElement,
+    length: number,
+    index: number,
+  ): void {
+    const stop = document.createElementNS('http://www.w3.org/2000/svg', 'stop');
+    const step = 100 / length;
+    const offset = index * step;
+    stop.setAttribute('offset', `${100 - offset}%`);
+    const rgb = `rgb(${color[0]},${color[1]},${color[2]})`;
+    stop.setAttribute('stop-color', rgb);
+
+    linearGradient.appendChild(stop);
+  }
+
+  /**
+   * Generate SVG gradient for selected color map. Used by either
+   * - interpolated layer legend or
+   * - color map select options
+   */
+  generateSVGGradientForColorMap(input: Layer<any> | number[][]) {
+    const forLayer = input instanceof Layer;
+    const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    svg.setAttribute('width', forLayer ? '75%' : '100%');
+    svg.setAttribute('height', '100%');
+
+    const defs = document.createElementNS('http://www.w3.org/2000/svg', 'defs');
+    const linearGradient = document.createElementNS(
+      'http://www.w3.org/2000/svg',
+      'linearGradient',
+    );
+
+    const id = `${+new Date()}-idwGradient-${
```

Comment:
interesting approach as Math.random() is already based on a pseudo-random generation based on a current date 🙂

Reply:
Just moved what was already used in  legend service and apparently back then when I implemented it I used date from some reason.  Will simplify this a bit.